Added basic coin functionality.

sjf60780 opened the pull request:
Adding coins to game

sjf60780 committed:
Added basic coin functionality. Coins can be red or yellow. Only the coin the player overlaps will be collected.

sjf60780 committed:
Added HUD, Brick, and coinIcon

sjf60780 committed:
Merge branch 'master' into coin

sjf60780 committed:
Adding HUD and images (bricks and coins)

sjf60780 committed:
Merge branch 'coin' of https://github.com/sjf60780/csc-303-platformer into coin

SamBumgardner commented:
Please resolve merge conflicts ASAP!

SamBumgardner commented:
Also, I'd like it if you could have picking up coins increment some variable in the Player class, and then use that Player class variable with the HeadsUpDisplay to display the player's current number of coins.

Right now the coins don't really do anything, but those two changes should make them feel more meaningful.

gmanjames commented on source/PlayState.hx:
```diff
@@ -20,6 +22,15 @@ class PlayState extends FlxState
+		//Coins are added to a group, coin group added to playstate
+		coins = new FlxGroup();
+		coins.add(Coin.createCoin(8, 8, true));
+		coins.add(Coin.createCoin(13, 8, true));
```
You might consider changing the third parameter for the Coin classes's constructor to be something like a FlxColor. That way it might be a little more clear as to what the third argument is doing here without having to go look inside of the Player class itself

gmanjames commented on source/Coin.hx:
```diff
@@ -0,0 +1,40 @@
+	 * @return
+	 */
+	static public function createCoin(X:Float=0, Y:Float=0, red:Bool=false):FlxSprite 
+	{
```
See comment above. Also, was there a particular reason that you use a static factory-type method (https://stackify.com/static-factory-methods/) like this as opposed to defining "new"?

gmanjames commented:
Those might be nit-picky things. I think it is looking good overall!

sjf60780 committed:
- Coins have been updated to where they are their own class.
- Player class now has variables to hold respective coin totals that are incremented upon collection

sjf60780 committed:
Merge branch 'master' into coin

sjf60780 commented:
I have updated the code with the wishes you both have requested. Coins are their own class now and as of right now the Player class has variables that will increment when the player collects a coin.

sjf60780 committed:
Update Player.hx
